> FAILED tests/test_executor.py::test_rx_calibration[mock] - RuntimeError: Rabi fit has chi2 2.013172369317522 greater than 2. Stopping.

@lballerio I thought you wanted to remove this check since it's meaningless

--- a/src/qibocal/protocols/two_qubit_interaction/cross_resonance/plotting.py
+++ b/src/qibocal/protocols/two_qubit_interaction/cross_resonance/plotting.py
@@ -142,26 +142,16 @@ def tomography_cr_plot(
             [
                 go.Scatter(
                     x=pair_data.x,
-                    y=bloch_vect_targ,
-                    name="Bloch vector |R(t)|",
-                    legendgroup="Bloch vector |R(t)|",
+                    y=y[0],
+                    name=f"{y[1]} Bloch vector |R(t)|",
+                    legendgroup=f"{y[1]} Bloch vector |R(t)|",
                     showlegend=True,
                     mode="markers",
                     marker=dict(
                         color="green",
                     ),
-                ),
-                go.Scatter(
-                    x=pair_data.x,
-                    y=bloch_vect_ctrl,
-                    name="Bloch vector |R(t)|",
-                    legendgroup="Bloch vector |R(t)|",
-                    showlegend=True,
-                    mode="markers",
-                    marker=dict(
-                        color="green",
-                    ),
-                ),
+                )
+                for y in [(bloch_vect_targ, "Target"), (bloch_vect_ctrl, "Control")]
             ],
             rows=[4, 4],
             cols=[1, 2],

--- a/src/qibocal/protocols/two_qubit_interaction/cross_resonance/plotting.py
+++ b/src/qibocal/protocols/two_qubit_interaction/cross_resonance/plotting.py
@@ -142,16 +142,19 @@ def tomography_cr_plot(
             [
                 go.Scatter(
                     x=pair_data.x,
-                    y=y[0],
-                    name=f"{y[1]} Bloch vector |R(t)|",
-                    legendgroup=f"{y[1]} Bloch vector |R(t)|",
+                    y=y,
+                    name=f"{label} Bloch vector |R(t)|",
+                    legendgroup=f"{label} Bloch vector |R(t)|",
                     showlegend=True,
                     mode="markers",
                     marker=dict(
                         color="green",
                     ),
                 )
-                for y in [(bloch_vect_targ, "Target"), (bloch_vect_ctrl, "Control")]
+                for y, label in [
+                    (bloch_vect_targ, "Target"),
+                    (bloch_vect_ctrl, "Control"),
+                ]
             ],
             rows=[4, 4],
             cols=[1, 2],

--- a/src/qibocal/protocols/two_qubit_interaction/cross_resonance/utils.py
+++ b/src/qibocal/protocols/two_qubit_interaction/cross_resonance/utils.py
@@ -30,15 +30,11 @@ def cross_resonance_pulses(
     cr_params = None
     canc_params = None
     if cnot_cal_seq is not None and len(cnot_cal_seq) != 0:
-        for p in cnot_cal_seq[2:]:
-            if cr_params is None and p[0] == cr_channel and isinstance(p[1], Pulse):
-                cr_params = p[1]
-            if (
-                canc_params is None
-                and p[1] == target_channel
-                and isinstance(p[1], Pulse)
-            ):
-                canc_params = p[1]
+        for ch, p in cnot_cal_seq[2:]:
+            if cr_params is None and ch == cr_channel and isinstance(p, Pulse):
+                cr_params = p
+            if canc_params is None and ch == target_channel and isinstance(p, Pulse):
+                canc_params = p
 
             if cr_params is not None and canc_params is not None:
                 # here we identified both the CR and cancellation pulse,

